src/workerd/jsg/setup.h

```diff
@@ -389,6 +389,25 @@ class Isolate: public IsolateBase {
           context, handle, jsg::TypeErrorContext::other());
     }
 
+    // Returns the constructor function for a given type declared as JSG_RESOURCE_TYPE.
+    //
+    // Note there's a useful property of class constructor functions: A constructor's __proto__
+    // is set to the parent type's constructor. Thus you can discover whether one class is a
+    // subclass of another by following the __proto__ chain.
+    //
+    // TODO(cleanup): This should return `JsFunction`, but there is no such type. We only have
+    //   `jsg::Function<...>` (or perhaps more appropriately, `jsg::Constructor<...>`), but we
+    //   don't actually know the function signature so that's not useful here. Should we add a
+    //   `JsFunction` that has no signature?
```

Comment:
Yeah, I intentionally didn't add `JsFunction` because I wanted make sure we thought through the relationship with `jsg::Function`. We could, for instance have something like `jsg::Function<JsValue(kj::ArrayPtr<JsValue> args)>` or we could have a generic `jsg::JsFunction`. Not sure which is the best approach. We don't need to solve that now, tho.